.github/workflows/CI.yml

```diff
@@ -21,66 +18,51 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout AD
-        uses: actions/checkout@v1
-
       - name: Setup Java ${{ matrix.java }}
         uses: actions/setup-java@v1
         with:
           java-version: ${{ matrix.java }}
 
-      - name: Run build
-        run: |
-          ./gradlew build
-          ls -ltr build/distributions/
-
-      - name: Multi Nodes Integration Testing
-        run: |
-          ./gradlew integTest  -PnumNodes=3
-
-      - name: Pull and Run Docker
-        run: |
-          plugin=`ls build/distributions/*.zip`
-          version=`echo $plugin|awk -F- '{print $4}'| cut -d. -f 1-3`
-          plugin_version=`echo $plugin|awk -F- '{print $4}'| cut -d. -f 1-4`
-          echo $version
-          cd ..
-
-          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$version
-          then
-            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$version" >> Dockerfile
+      # dependencies: OpenSearch
+      - name: Checkout OpenSearch
+        uses: actions/checkout@v2
+        with:
+          repository: 'opensearch-project/OpenSearch'
+          path: OpenSearch
+      - name: Build OpenSearch
+        working-directory: ./OpenSearch
+        run: ./gradlew publishToMavenLocal
 
-            ## The ESRestTest Client uses http by default.
-            ## Need to disable the security plugin to call the rest api over http.
-            echo "RUN if [ -d /usr/share/opensearch/plugins/opendistro-anomaly-detection ]; then /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro-anomaly-detection; fi" >> Dockerfile
-            echo "ADD anomaly-detection/build/distributions/opendistro-anomaly-detection-$plugin_version.zip /tmp/" >> Dockerfile
-            echo "RUN /usr/share/opensearch/bin/elasticsearch-plugin install --batch file:/tmp/opendistro-anomaly-detection-$plugin_version.zip" >> Dockerfile
+      # dependencies: common-utils
+      - name: Checkout common-utils
+        uses: actions/checkout@v2
+        with:
+          repository: 'dblock/common-utils'
```

Comment:
Minor: can we avoid using a personal repository name?
Ignore this comment if this is just WIP/temporary solution/testing purpose.

Reply:
This is https://github.com/opensearch-project/common-utils/pull/4 that would need to be merged.

Reply:
Got it. I mean it's good to remove this personal S3 maven repo `url = 's3://search-vemsarat/'`, but seems we use another personal local path with name `'dblock/...`.  

Currently it's not convenient to build/test locally. Will we publish all dependencies to public maven soon?

Reply:
The personal path is just pending PRs being merged if we decide to go this route.

We'll (I'll) evaluate jitpack to compare. It currently doesn't "just work", needs to successfully be able to check out and build these dependencies.

Reply:
+1, once we evaluate jitpack, we should make changes to common-utils and job-scheduler, build them locally here in the workflows instead of fork repositories.